frontend/src/views/ShareSecretPublicPage/ShareSecretPublicPage.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useMemo, useState } from "react";
+import Head from "next/head";
+import Image from "next/image";
+import Link from "next/link";
+import { useRouter } from "next/router";
+
+import { openSignedAssymmetric } from "@app/components/utilities/cryptography/crypto";
+import { useToggle } from "@app/hooks";
+import { useGetActiveSharedSecretById } from "@app/hooks/api/secretSharing";
+
+import { DragonMainImage, SecretTable } from "./components";
+
+export const ShareSecretPublicPage = () => {
+  const router = useRouter();
+  const { id, key: urlEncodedPublicKey } = router.query;
+
+  const publicKey = decodeURIComponent(urlEncodedPublicKey as string);
+  useEffect(() => {
+    if (!id || !publicKey) {
+      router.push("/404");
+    }
+  }, [id, publicKey]);
+
+  const { isLoading, data } = useGetActiveSharedSecretById(id as string);
```

Comment:
Currently the Shared URL looks something like this:
```sh
http://localhost:8080/shared/secret/1f723b86-2d76-4f6c-8628-ea9a8b9da63a?key=UEWk%2FMmWc7Nfx6m9uYo0sbMc68DzApZmtbNlsO3s98Q%3D
```
so a user would need to get the combination of the UUID as well as the key right to get the decoded data otherwise we issue an invalid URL error.

With the hashing, I think we will be able to limit the brute-force attack more prominently, is that what you meant? And then we'd need to store the hash in the DB too right?